Extensions/MUC-Sub/XMPPMUCSub.h

```diff
@@ -0,0 +1,272 @@
+//
+//  XMPPMUCSub.h
+//  XMPPFramework
+//
+//  Created by Robert Lohr on 06.10.2017.
+//
+
+#import <Foundation/Foundation.h>
+#import "XMPPModule.h"
+
+#ifndef MUCSub_h
+#define MUCSub_h
+
+@class XMPPIDTracker;
+@class XMPPIQ;
+@class XMPPJID;
+@class XMPPMessage;
+@class XMPPPresence;
+@class XMPPRoom;
+
+/**
+ * The XMPPMUCSub provides functionality to a proprietary Multi User Chat extension of 
+ * the ejabberd XMPP server. This extension aims to provide a solution to the problem
+ * that users are required to send a presence to a MUC room in oder to receive messages.
+ * By subscribing to the room a user can also participate if not online. Once reconnected,
+ * missed messages are synced.
+ * 
+ * The extension leverages several existing extension to achieve its task. More details
+ * can be found on the project's website (as of 09.10.2017).
+ * 
+ * https://docs.ejabberd.im/developer/xmpp-clients-bots/proposed-extensions/muc-sub/
+ * 
+ * One note about subscriptions (taken from the ejabberd documentation):
+ * Subscription is associated with a nick. It will implicitly register the nick. Server 
+ * should otherwise make sure that subscription match the user registered nickname in 
+ * that room. In order to change the nick and/or subscription nodes, the same request 
+ * MUST be sent with a different nick or nodes information.
+ *
+ * This means that clients need to provide the nickname of the user in the MUC room
+ * when subscribing. If none is given then the bare JID will be used.
+ * 
+ * MUC-Sub can be enabled by creating an instance of `XMPPMUCSub` and activating it on 
+ * the `XMPPStream`.
+**/
+@interface XMPPMUCSub : XMPPModule
+{
+    XMPPIDTracker *xmppIDTracker;
+}
+
+// MARK: Service Discovery
+
+/**
+ * Query whether MUC-Sub is enabled on a room.
+ * 
+ * @param room
+ *        The `XMPPRoom` for which to check if MUC-Sub has been enabled.
+ * 
+ * @return
+ * The request id of the IQ is in case client code may want to do manual tracking. `nil`
```

Comment:
"Returns the randomly generated "id" attribute value of the standard <iq> element that 
is sent in case client code may want to do manual tracking."?

Reply:
If there's no precedence in XMPPFramework, let's stick to the wording in [the RFC](https://xmpp.org/rfcs/rfc6120.html#stanzas-attributes-id):

"'id' attribute of the IQ stanza that was sent to the server" and similar